core/test/config/config.cpp

```diff
@@ -0,0 +1,194 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <ginkgo/core/config/config.hpp>
+
+
+#include <typeinfo>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/solver/bicg.hpp>
+#include <ginkgo/core/solver/cg.hpp>
+#include <ginkgo/core/stop/combined.hpp>
+#include <ginkgo/core/stop/iteration.hpp>
+#include <ginkgo/core/stop/residual_norm.hpp>
+
+
+#include "core/config/config_helper.hpp"
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+
+using namespace gko::config;
+
+
+class Config : public ::testing::Test {
+protected:
+    using value_type = double;
+    using Mtx = gko::matrix::Dense<value_type>;
+    Config()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(gko::initialize<Mtx>(
+              {{2, -1.0, 0.0}, {-1.0, 2, -1.0}, {0.0, -1.0, 2}}, exec)),
+          stop_config({{"Type", pnode{"Iteration"}}, {"max_iters", pnode{1}}})
+    {}
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::shared_ptr<Mtx> mtx;
+    pnode stop_config;
+};
```

Comment:
could you give more on it? I would like to use a complete test to ensure everything working. the cg test may be moved to another one when we have the solver config.

Reply:
I think this comment might be misplaced. It's meant for the `GenerateMap` test. That test only checks for non-throwing, but that seems pointless to me. It would probably need to check which keys are in the map, but checking the values seems to be quite tough.